### ISSUE TITLE
Execute Custom Script Before Applying Conventional Commits

### ISSUE THREAD

**Issue by theovier** [DanySK/gradle-pre-commit-git-hooks#517]
### What would you like to be able to do?

Hi,

I want to modify my commit message by a custom script (basically extracting a Jira ticket number from the branch name and prefixing the commit with it) and then still use your check for conventional commits afterwards. 

I couldn't find a way to chain instructions for the `commitMsg`. I could only do either or. I could write all the logic for the convential commits inside my custom script too but this would defeat the purpose.

I tried

```gradle
    commitMsg {
        conventionalCommits()
        appendScript { """            
            ./scripts/prepare-commit-msg.sh
            """ }
    }
```
but to no avail.

Kind regards,
Theo

### Is this a feature you are interested in implementing yourself?

No, I do not have the skills yet to do it

**Comment by DanySK** [comment]
Should be supported already. Can you produce a minimal reproduction kit?

**Comment by theovier** [comment]
Yes, however, it will probably take some time. Chances are good that you really do support it already and I just couldn't wrap my head around it. 

Just to check: the idea of executing another script in the `appendScript` is the way you thought it could be done too, right?

**Comment by DanySK** [comment]
Yes

**Comment by DanySK** [comment]
Uhmm... try this instead:

```
    commitMsg {
        conventionalCommits()
        appendScript {
            File("./scripts/prepare-commit-msg.sh").readText()
         }
    }
```

I'm closing as there's been no interaction for over one year. Feel free to reopen if the issue still stands.
